lib/index.ts

```diff
@@ -115,9 +117,17 @@ export async function inspect(
     options = { dev: false, scanAllUnmanaged: false };
   }
 
+  if (!snykHttpClient) {
+    throw new Error('No HTTP client provided!');
```

Comment:
is there any case in which it's ok not to have a HTTP client? like scanning a pom.xml for example
maybe we move this inside the ifs below where a client if definitely required

Reply:
or inside the `createDepGraphFromArchive` function

Reply:
Done.